Makefile

```
@@ -1,185 +1,92 @@
 
-# define VERBOSE to see executed commands
 # default build configuration
-TARGET=hyped
-MAIN=run/main.cpp
-CROSS=0
-NOLINT=0
+# TARGET  - define the binary name to compile into
+# MAIN    - define application entry point
+# CROSS   - set to 1 to use a cross-compiler, to compile on a laptop and run on BBB
+# NOLINT  - set to 1 to prevent linting the code
+# VERBOSE - set to 1 to print all commands Makefile runs
+TARGET  := hyped
+MAIN    := run/main.cpp
+CROSS   := 0
+NOLINT  := 0
+VERBOSE := 0
+RELEASE := 0
+
+# include helper files
+HELPER  := utils/build
+include $(HELPER)/config.mk
+include $(HELPER)/build.mk
+include $(HELPER)/libs.mk
+include $(HELPER)/test.mk
```

```suggestion
HELPERS_DIR  := utils/build
include $(HELPERS_DIR)/config.mk
include $(HELPERS_DIR)/build.mk
include $(HELPERS_DIR)/libs.mk
include $(HELPERS_DIR)/test.mk
```
Not v important but I read this section 3 times before fully understanding it. (At a first glance, I knew it was importing the other makefiles. I thought the `HELPER` variable was unrelated. On second reading, I thought you used `HELPER` just to avoid typing `utils/build` many times. Then I noticed that you actually call the other makefiles helpers.)